### PR TITLE
[Merged by Bors] - feat: add `Filter.Eventually.self_of_nhdsSet`

### PR DESCRIPTION
**Pull request by urkud**
Also add `subset_of_mem_nhdsSet`.
From the Sphere Eversion Project.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/NhdsSet.lean [hunk 1]
```diff
@@ -80,6 +80,13 @@ theorem principal_le_nhdsSet : 𝓟 s ≤ 𝓝ˢ s := fun _s hs =>
   (subset_interior_iff_mem_nhdsSet.mpr hs).trans interior_subset
 #align principal_le_nhds_set principal_le_nhdsSet
 
+theorem subset_of_mem_nhdsSet (h : t ∈ 𝓝ˢ s) : s ⊆ t := principal_le_nhdsSet h
+
+theorem Filter.Eventually.on_set {p : α → Prop} (h : ∀ᶠ x in 𝓝ˢ s, p x) : ∀ x ∈ s, p x :=
```
Comment (ADedecker) [inline Mathlib/Topology/NhdsSet.lean:85]: What do you think of `Filter.Eventually.self_of_nhdsSet`, to match [Filter.Eventually.self_of_nhds](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Basic.html#Filter.Eventually.self_of_nhds)?

Comment (urkud) [reply Mathlib/Topology/NhdsSet.lean:85]: @PatrickMassot added this to the sphere eversion project. I don't care about the name.

**Comment by ocfnash** [comment]
Thanks!

bors merge

**Comment by ADedecker** [comment]
@urkud Were we waiting for Patrick to approve the name?

**Comment by urkud** [comment]
> @urkud Were we waiting for Patrick to approve the name?

It wasn't me who merged it. We can always rename later.

**Pull request closed without merging**